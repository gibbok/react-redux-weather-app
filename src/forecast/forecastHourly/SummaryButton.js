// @flow
import * as React from 'react'
import {FlatButton} from 'material-ui'

/* eslint-disable no-undef */
type PropsType = {
  +onSummaryClick: Function
}
/* eslint-enable no-undef */

const SummaryButton:React.StatelessFunctionalComponent<any> = ({ onSummaryClick }:PropsType):React.Element<any> => {
  return (
    <FlatButton onClick={(e) => onSummaryClick(e)}>Summary</FlatButton>
  )
}

export default SummaryButton
