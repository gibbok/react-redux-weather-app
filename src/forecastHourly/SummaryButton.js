import React from 'react'

const SummaryButton = ({ onSummaryClick }) => {
  return (
    <button type='button' onClick={(e) => onSummaryClick(e)}>Summary</button>
  )
}

export default SummaryButton
