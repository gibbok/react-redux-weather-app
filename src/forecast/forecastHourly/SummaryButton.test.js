import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import sinon from 'sinon'
import SummaryButton from './SummaryButton'

describe('<SummaryButton />', () => {
  it('should render', () => {
    const wrapper = shallow(<SummaryButton />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('should simulate click event', () => {
    const onSummaryClick = sinon.spy()
    let wrapper = shallow(<SummaryButton onSummaryClick={onSummaryClick} />)
    wrapper.find('button').simulate('click')
    expect(onSummaryClick.called).toEqual(true)
  })
})
