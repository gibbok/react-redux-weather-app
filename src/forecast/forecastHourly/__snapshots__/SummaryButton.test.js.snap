// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryButton /> should render 1`] = `
<withStyles(Button)
  onClick={[Function]}
>
  Summary
</withStyles(Button)>
`;
